c++/src/capnp/test-util.c++

```diff
@@ -1192,6 +1192,13 @@ kj::Promise<void> TestMoreStuffImpl::throwRemoteException(ThrowRemoteExceptionCo
   return KJ_EXCEPTION(FAILED, "remote exception: test exception");
 }
 
+kj::Promise<void> TestMoreStuffImpl::throwExceptionWithDetail(
+    ThrowExceptionWithDetailContext context) {
+  auto e = KJ_EXCEPTION(FAILED, "test exception");
+  e.setDetail(1, kj::heapArray<byte>(0));
```

Comment:
We should probably test a non-empty detail to make sure the bytes are in fact encoded correctly?